python/xorbits/_mars/dataframe/datasource/read_parquet.py

```diff
@@ -395,33 +402,64 @@ def _tile_no_partitioned(cls, op: "DataFrameReadParquet"):
 
         dtypes = cls._to_arrow_dtypes(out_df.dtypes, op)
         shape = (np.nan, out_df.shape[1])
-
         path_prefix = ""
+        z = None
         if isinstance(op.path, (tuple, list)):
             paths = op.path
-        elif get_fs(op.path, op.storage_options).isdir(op.path):
+        elif fsspec.get_fs_token_paths(op.path, storage_options=op.storage_options)[
+            0
+        ].isdir(op.path):
             parsed_path = urlparse(op.path)
             if parsed_path.scheme.lower() == "hdfs":
                 path_prefix = f"{parsed_path.scheme}://{parsed_path.netloc}"
-            paths = get_fs(op.path, op.storage_options).ls(op.path)
+            paths = fsspec.get_fs_token_paths(
+                op.path, storage_options=op.storage_options
+            )[0].ls(op.path)
+            paths = sorted(paths)
+        elif isinstance(op.path, str) and op.path.endswith(".zip"):
+            z = zipfile.ZipFile(op.path)
+            paths = z.namelist()
+            paths = [
+                path
+                for path in paths
+                if path.endswith(".parquet") and not path.startswith("__MACOSX")
+            ]
         else:
-            paths = glob(op.path, storage_options=op.storage_options)
-
+            if op.path.startswith("hdfs"):
```

Comment:
fsspec implements the hdfs protocol: https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.implementations.arrow.HadoopFileSystem
Why not use `fsspec.get_fs_token_paths`?

Reply:
`HadoopFileSystem.glob` will delete the host and port of path. For example, when the `op.path` is `hdfs://localhost:8020/tmp/test/test.zip` , the output of `HadoopFileSystem.glob` will be `/tmp/test/test.zip` which will cause error `file not found` subsequently

Reply:
`fsspect.get_fs_token_paths` always strip protocol from the input path, not only for the hdfs backend.